windowsforms/FlexGridShowcaseDemo/cs/MainForm.cs

```diff
@@ -0,0 +1,682 @@
+                .Where(x => !string.IsNullOrEmpty(x.Name))
+                .Select(x => new RibbonToggleButton()
+                {
+                    Text = x.Caption,
+                    Pressed = x.Visible
+                });
+
+            foreach (var columnRibbonToggleButton in columnRibbonToggleButtons)
+            {
+                columnRibbonToggleButton.PressedButtonChanged += ColumnRibbonToggleButton_PressedButtonChanged;
+                _ribbonMenuColumns.Items.Add(columnRibbonToggleButton);
+            }
+
+            _ribbonComboBoxDataSize.SelectedIndex = 2;
+
+            ActiveControl = _flexGrid;
+        }
+
+        #endregion
+
+        #region Fill Data
+
+        private enum Countries
+        {
+            Germany,
+            UK,
+            US,
+            Japan
+        };
+
+        private enum DrawColors
+        {
+            Black,
+            White,
+            Green,
+            Red
+        };
+
+        private void BuildRows(int rowCount, DataTable dataTable)
+        {
+            var startPeriod = new DateTime(2020, 01, 25, 8, 29, 0);
+            _flexGrid.BeginUpdate();
+
+            // Related data
+            var products = (from s in _dataSet.Tables["Products"].Rows.Cast<DataRow>() select s)
+                    .Select(x => x["Name"].ToString())
+                    .ToArray();
+
+            var countries = Enum.GetValues(typeof(Countries))
+                    .Cast<Countries>()
+                    .ToArray();
+
+            var colors = Enum.GetValues(typeof(DrawColors))
+                    .Cast<DrawColors>()
+                    .ToArray();
+
+            dataTable.Clear();
+
+            var dataRows = dataTable.Rows;
+            for (var rowIndex = 0; rowIndex < rowCount; rowIndex++)
+            {
+                // Fill history data
+                var historyData = new int[12];
+                for (var i = 0; i < historyData.Length; i++)
+                {
+                    historyData[i] = _rnd.Next(0, 50);
+                };
+
+                dataRows.Add(
+                    // ID
+                    rowIndex + 1,
+                    // Product
+                    products[_rnd.Next(products.Length)],
+                    // Country
+                    countries[_rnd.Next(countries.Length)],
+                    // Color
+                    colors[_rnd.Next(colors.Length)],
+                    // Price
+                    _rnd.NextDouble() * 100 * _rnd.Next(100),
+                    // Change
+                    _rnd.NextDouble() * 10 * _rnd.Next(100) * (_rnd.Next(1, 3) == 1 ? (-1) : 1),
+                    // History
+                    historyData,
+                    // Discount
+                    _rnd.NextDouble(),
+                    // Rating
+                    _rnd.Next(0, 5),
+                    // Active
+                    (_rnd.Next(1, 3) == 1 ? false : true),
```

Comment:
Probably `_rnd.NextDouble() >= 0.5` would be simpler to generate a random boolean.

```suggestion
                    (_rnd.NextDouble() >= 0.5),
```

Reply:
Fixed with b4f0b07ed0a6576b5bf1d80ef011e2a3c8ded91b.